crates/bevy_asset/src/io/memory.rs

```diff
@@ -236,6 +237,31 @@ impl AsyncRead for DataReader {
     }
 }
 
+impl AsyncSeek for DataReader {
+    fn poll_seek(
+        mut self: Pin<&mut Self>,
+        _cx: &mut std::task::Context<'_>,
+        pos: SeekFrom,
+    ) -> Poll<std::io::Result<u64>> {
+        let new_pos = match pos {
+            SeekFrom::Start(offset) => offset as i64,
```

Comment:
Let's try to avoid `as` conversions here. It's best to use a TryInto conversion and return it as a result on failure.

Reply:
Okay, I replaced all `as` conversions with `try_into` calls.